torch/csrc/distributed/c10d/TCPStoreLibUvBackend.cpp

```diff
@@ -1407,6 +1470,43 @@ void LibUVStoreDaemon::wakeupWaitingClients(const std::string& key) {
   }
 }
 
+void LibUVStoreDaemon::wakeupOneWaitingClient(const std::string& key) {
```

Comment:
Just n00b question, why we don't have this for other operations within libuv backend?

Reply:
Normally with set/wait all of the waiting clients get woken up. With queues we only want a single worker to wake up and try and get the item so that's why we need a special "wakeupOne"